### PR TITLE
FIN-13060 up to php 8

### PR DESCRIPTION
**Pull request by sergiizeleniy**
ISSUES RESOLVED
-------
https://finsight.myjetbrains.com/youtrack/issue/FIN-13060
https://finsight.myjetbrains.com/youtrack/issue/FIN-13939

### CODE REVIEW

**Review by denys-filchakov — APPROVED** [review]

**Review by YevhenKyrychenko — APPROVED** [review]

File: .github/workflows/psalm-and-phpunit.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
-          php-version: '7.4.19'
+          php-version: '8.0.13'
```
Comment (bogdankoliesnik) [inline .github/workflows/psalm-and-phpunit.yml:15]: let's use 8.1 as in a psalm config

Comment (sergiizeleniy) [reply .github/workflows/psalm-and-phpunit.yml]: done

**Review by bogdankoliesnik — APPROVED** [review]

**Review by YevhenKyrychenko — APPROVED** [review]

File: lib/Map/ObjectMap.php [hunk 2]
```diff
@@ -35,52 +36,45 @@
...
     {
         $valueType = null;
-        $reflection = new \ReflectionClass(static::class);
+        $reflection = new ReflectionClass(static::class);
         $fake = $reflection->newInstanceWithoutConstructor();
-        if (is_a(static::class, StrictTypedValue::class, true)) {
-            /** @var StrictTypedValue $fake */
-            $valueType = $fake->getValueType();
-        }
+        //if (is_a(static::class, StrictTypedValue::class, true)) {
```
Comment (bogdankoliesnik) [inline lib/Map/ObjectMap.php:67]: missed?

File: lib/Enum.php [hunk 3]
```diff
@@ -76,13 +75,10 @@ final public function equalTo(Enum $another): bool
...
 
-    /**
-     * @return static
-     */
-    final public static function __callStatic(string $name, $_)
+    final public static function __callStatic(string $name, array $arguments): static
     {
-        $arguments = self::getConstantArguments($name);
-        if (!$arguments) {
+        $arguments_ = self::getConstantArguments($name);
```
Comment (bogdankoliesnik) [inline lib/Enum.php:80]: nit: weird naming with `_`

File: lib/Enum.php [hunk 4]
```diff
@@ -61,7 +60,7 @@ final private function __construct(string $value, array $arguments = null)
         }
     }
 
-    final private function __clone()
+    private function __clone()
```
Comment (bogdankoliesnik) [inline lib/Enum.php:63]: was it changed for purpose?

**Review by rsataev — APPROVED** [review]

File: lib/Map/ObjectMap.php (hunk 2 above)
Comment (sergiizeleniy) [reply lib/Map/ObjectMap.php:67]: It was cleaning up after @alexander-masko work
https://github.com/dealroadshow/util/blob/02ca98f9d7057343b99161912e6061ea172e46b2/lib/Map/ObjectMap.php#L33

File: lib/Enum.php (hunk 3 above)
Comment (sergiizeleniy) [reply lib/Enum.php:80]: https://www.php.net/manual/en/migration80.incompatible.php
...
Magic Methods will now have their arguments and return types checked if they have them declared. The signatures should match the following list:
...
__callStatic(string $name, array $arguments): mixed
...
You can suggest another name!

File: lib/Enum.php (hunk 4 above)
Comment (sergiizeleniy) [reply lib/Enum.php:63]: https://www.php.net/manual/en/migration80.incompatible.php
...
Applying the final modifier on a private method will now produce a warning unless that method is the constructor.
....

File: lib/Enum.php (hunk 3 above)
Comment (bogdankoliesnik) [reply lib/Enum.php:80]: tricky stuff)

**Review by bogdankoliesnik — APPROVED** [review]

**Review by denys-filchakov — APPROVED** [review]

**Pull request merged**